src/editor.rs

```diff
@@ -7,335 +7,337 @@ use std::collections::hash_map::DefaultHasher;
+    
+        use super::ActionCategory;
+        use tuikit::prelude::*;
+    
+        /// Returns the [`Color`] that all [`Action`]s of a given [`ActionCategory`] should be
+        /// displayed.  This is not implemented as a method on [`ActionCategory`], because doing so
+        /// would require [`ActionCategory`] to rely on the specific terminal backend used.  This way,
+        /// we keep the terminal backend as encapsulated as possible.
+        pub fn term_color(category: ActionCategory) -> Color {
+            match category {
+                ActionCategory::Move => Color::LIGHT_BLUE,
+                ActionCategory::History => Color::LIGHT_YELLOW,
+                ActionCategory::Insert => Color::LIGHT_GREEN,
+                ActionCategory::Replace => Color::CYAN,
+                ActionCategory::Delete => Color::RED,
+                ActionCategory::Quit => Color::MAGENTA,
+                ActionCategory::Undefined => Color::LIGHT_RED,
             }
         }
-
-        /// Sets and enforces the max entry limit
-        pub fn set_max_entries(&mut self, max_entries: usize) {
-            self.max_entries = max_entries;
-            self.enforce_entry_limit();
+    
+        /// One entry in the log.  This usually represts a single command, but could represent an
+        /// accumulation of many identical commands that are executed consecutively.
+        struct Entry {
+            count: usize,
+            command: String,
+            description: String,
+            color: Color,
         }
-
-        /// Draw a log of recent commands to a given terminal at a given location
-        pub fn render(&self, term: &Term, row: usize, col: usize) {
-            // Calculate how wide the numbers column should be, enforcing that it is at least two
-            // chars wide.
-            let count_col_width = self
-                .commands
-                .iter()
-                .map(|e| match e.count {
-                    1 => 0,
-                    c => format!("{}x", c).len(),
-                })
-                .max()
-                .unwrap_or(0)
-                .max(2);
-            // Calculate the width of the command column, and make sure that it is at least two
-            // chars wide.
-            let cmd_col_width = self
-                .commands
-                .iter()
-                .map(|e| e.command.len())
-                .max()
-                .unwrap_or(0)
-                .max(2);
-            // Render the commands
-            for (i, e) in self.commands.iter().enumerate() {
-                // Print the count if greater than 1
-                if e.count > 1 {
-                    term.print(row + i, col, &format!("{}x", e.count)).unwrap();
+    
+        /// A utility struct to store and display a log of which commands have been executed recently.
+        /// This is mostly used to give the viewers of my streams feedback for what I'm typing.
+        pub struct CommandLog {
```

Comment:
This should probably be called `KeyStrokeLog`, on the basis that the `Command` enum has also been renamed...

Reply:
Do I need to change the `mod command_log` to `mod keystroke_log` too?